server/controllers/user/search.js

```diff
@@ -0,0 +1,88 @@
+const { Op, fn, col } = require('sequelize');
+
+const { searchRequestValidation } = require('../../utils/validation');
+const { customError } = require('../errors');
+
+const {
+  Location,
+  User,
+  SubServices,
+  Review,
+  MainServices,
+} = require('../../database/models');
+
+const getSearchResult = async (req, res, next) => {
+  try {
+    const {
+      name, location, service, subservice, page = 1,
+    } = await searchRequestValidation.validateAsync(req.query);
+    const where = {};
+    if (name) {
+      where[Op.or] = [
+        {
+          first_name: {
+            [Op.iLike]: `%${name}%`,
+          },
+        },
+        {
+          last_name: {
+            [Op.iLike]: `%${name}%`,
+          },
+        },
+      ];
+    }
+    if (location) {
+      where.locationId = location;
+    }
+    if (subservice) {
+      const subServiceArray = subservice?.split(',');
+      where['$services.id$'] = subServiceArray;
+    }
+    if (service) {
+      where.mainServiceId = +service;
+    }
+    const { count, rows } = await User.findAndCountAll({
+      limit: 6,
+      offset: (page - 1) * 6,
+      where,
+      include: [
+        { model: Location },
```

Comment:
Select attribuates you need only for all models
```suggestion
        { model: Location,  attributes: ['city'] },
```